### PR TITLE
Add Shell Component to Design System 

### PR DESCRIPTION
**Pull request by dzhavat**
## Description

* Add Shell component to Watt Design System
* Use Shell component in DataHub app
* Create `dh-app/core/feature-shell` library
* Create Storybook page where the component is showcased
* Add basic Nx tags configuration

## References

- Energinet-DataHub/MightyDucks#132

### CODE REVIEW

File: libs/dh-app/core/feature-shell/src/lib/dh-app-core-feature-shell.module.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { NgModule } from '@angular/core';
+
+import { DhAppCoreFeatureShellRoutingModule } from './dh-app-core-feature-shell-routing.module';
+import { ShellModule } from './shell/shell.module';
+
+@NgModule({
+  imports: [ShellModule, DhAppCoreFeatureShellRoutingModule],
+})
+export class DhAppCoreFeatureShellModule {}
```
Comment (LayZeeDK) [inline libs/dh-app/core/feature-shell/src/lib/dh-app-core-feature-shell.module.ts:9]: suggestion: a feature shell module contains root level routes, so we use `RouterModule.forRoot`
suggestion: make the feature shell idiomatic and simpler by deleting the separate routing module

```suggestion
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { ShellComponent } from './shell/shell.component';
import { ShellModule } from './shell/shell.module';

const routes: Routes = [
  {
    path: '',
    component: ShellComponent,
    children: [
      // Lazy feature routes
    ],
  },
];

@NgModule({
  exports: [RouterModule],
  imports: [RouterModule.forRoot(routes), ShellModule],
})
export class DhAppCoreFeatureShellModule {}
```

File: libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    <p>Sidenav</p>
+  </ng-container>
+
+  <ng-container toolbar>
+    <p>Toolbar</p>
+  </ng-container>
+
+  <h1>THIS IS THE APP</h1>
+  <watt-autocomplete></watt-autocomplete>
+</watt-shell>
```
Comment (LayZeeDK) [inline libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html:12]: suggestion: add router outlet for app features

```suggestion
  <router-outlet></router-outlet>
</watt-shell>
```

File: libs/dh-app/core/feature-shell/src/lib/shell/shell.module.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { NgModule } from '@angular/core';
+import { WattModule } from '@energinet/watt';
+
+import { ShellComponent } from './shell.component';
+
+@NgModule({
+  declarations: [ShellComponent],
+  imports: [WattModule],
+})
+export class ShellModule {}
```
Comment (LayZeeDK) [inline libs/dh-app/core/feature-shell/src/lib/shell/shell.module.ts:10]: suggestion: support router outlet for app features

```suggestion
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { WattModule } from '@energinet/watt';

import { ShellComponent } from './shell.component';

@NgModule({
  declarations: [ShellComponent],
  imports: [RouterModule, WattModule],
})
export class ShellModule {}

```

File: libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<watt-shell>
+  <ng-container sidenav>
```
Comment (LayZeeDK) [inline libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html:2]: suggestion: use a `watt-shell` prefix to be explicit about cohesive parts

```suggestion
  <ng-container watt-shell-sidenav>
```

File: libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    <p>Sidenav</p>
+  </ng-container>
+
+  <ng-container toolbar>
```
Comment (LayZeeDK) [inline libs/dh-app/core/feature-shell/src/lib/shell/shell.component.html:6]: suggestion: use a `watt-shell` prefix to be explicit about cohesive parts

```suggestion
  <ng-container watt-shell-toolbar>
```

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

**Review by frontend-specialisten — APPROVED** [review]
I think this looks good as an MVP. When we have the colors, breakpoints, sizes and theming we can apply some default styles and add open / close for the sidebar. 
Also I think we should create a task for sidebar navigation items, which are styled by our design

**Comment by dzhavat** [comment]
I've omitted anything related to spacing, theming, open/close functionality because these are not defined yet. There are probably some commonly used ways to toggle a sidebar, which we can also add, but I haven't seen how our Design System handles it. That's why it's yet to be implemented.

**Comment by dzhavat** [comment]
@LayZeeDK Do you see the need for you reviewing this PR as well?

**Review by LayZeeDK — APPROVED** [review]

**Pull request merged**